src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Kusto.Data.Common;
+using Kusto.Data.Net.Client;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.DotNet.Kusto
+{
+    public class KustoClientProvider : IKustoClientProvider
+    {
+        private readonly IOptions<KustoClientProviderOptions> _options;
+
+        public KustoClientProvider(IOptions<KustoClientProviderOptions> options)
+        {
+            _options = options;
+        }
+
+        public KustoClientProviderOptions Options => _options.Value;
```

Comment:
I wouldn't make this public.  No one outside of this class has any reason to be mucking around with your options.